### PR TITLE
Clarify slots documentation

### PR DESCRIPTION
**Pull request by nathanl**
- Expand the explanations a bit
- Fix: pass `entries` in the pattern matching example
- Show that `:let` defines the variable name for the block
-  Show how to conditionally render a slot

### CODE REVIEW

File: lib/phoenix_component.ex [hunk 1]
```diff
@@ -150,12 +155,18 @@ defmodule Phoenix.Component do
...
       end
 
-  And now you can invoke it as:
+  When calling the component, you can use the special attribute `:let` (note
+  the leading `:`) to take the value that the component passes back and bind it
+  to a variable:
 
-      <.unordered_list :let={entry} entries={~w(apple banana cherry)}>
-        I like <%= entry %>
+      <.unordered_list :let={an_entry} entries={~w(apple banana cherry)}>
```
Comment (nathanl) [inline lib/phoenix_component.ex:162]: Maybe it was just me, but I needed to experiment to see for sure that `:let` determines what the variable name is, so making it slightly different here shows that this is the case.

**Comment by nathanl** [comment]
Thanks to @schrockwell for suggesting the conditional trick and @chrismccord for confirming

File: lib/phoenix_component.ex [hunk 2]
```diff
@@ -285,6 +298,44 @@ defmodule Phoenix.Component do
...
+  You can't wrap a conditional around a slot because a slot entry must be
+  a direct child of a component.
+  But you can accomplish this by passing an attribute.
+
+      <.table rows={@users}>
+        <:col :let={user} label="Name">
+          <%= user.name %>
+        </:col>
+
+        <:col :let={user} label="Address" if={ @current_user.role == :admin }>
```
Comment (josevalim) [inline lib/phoenix_component.ex:312]:
```suggestion
        <:col :let={user} label="Address" if={@current_user.role == :admin}>
```

File: lib/phoenix_component.ex [hunk 3]
```diff
@@ -285,6 +298,44 @@ defmodule Phoenix.Component do
...
+        <:col :let={user} label="Name">
+          <%= user.name %>
+        </:col>
+
+        <:col :let={user} label="Address" if={@current_user.role == :admin}>
+          <%= user.address %>
+        </:col>
+      </.table>
+
+
```
Comment (josevalim) [inline lib/phoenix_component.ex:317]:
```suggestion
```

File: lib/phoenix_component.ex [hunk 4]
```diff
@@ -285,6 +298,44 @@ defmodule Phoenix.Component do
...
+  Then you can check for that attribute within the component:
+
+      def table(assigns) do
+        ~H"""
+        <table>
+          <tr>
+            <%= for col <- @col, col[:if] != false do %>
```
Comment (josevalim) [inline lib/phoenix_component.ex:324]:
```suggestion
            <%= for col <- @col, Map.get(col, :if, true) do %>
```

File: lib/phoenix_component.ex [hunk 5]
```diff
@@ -285,6 +298,44 @@ defmodule Phoenix.Component do
...
+              <th><%= col.label %></th>
+            <% end %>
+          </tr>
+          <%= for row <- @rows do %>
+            <tr>
+              <%= for col <- @col, col[:if] != false do %>
```
Comment (josevalim) [inline lib/phoenix_component.ex:330]:
```suggestion
              <%= for col <- @col, Map.get(col, :if, true) do %>
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**